src/ioctls/vm.rs

```diff
@@ -358,11 +389,11 @@ mod tests {
         let evtfd1 = unsafe { eventfd(0, EFD_NONBLOCK) };
         let evtfd2 = unsafe { eventfd(0, EFD_NONBLOCK) };
         let evtfd3 = unsafe { eventfd(0, EFD_NONBLOCK) };
-
-        assert!(vm_fd.register_irqfd(evtfd1, 4).is_ok());
-        assert!(vm_fd.register_irqfd(evtfd2, 8).is_ok());
-        assert!(vm_fd.register_irqfd(evtfd3, 4).is_ok());
-
+        if cfg!(any(target_arch = "x86", target_arch = "x86_64")) {
+            assert!(vm_fd.register_irqfd(evtfd1, 4).is_ok());
+            assert!(vm_fd.register_irqfd(evtfd2, 8).is_ok());
+            assert!(vm_fd.register_irqfd(evtfd3, 4).is_ok());
+        }
```

Comment:
With this change the following asserts fail for different reasons on arm and x86. I think it is worth mentioning why we expect the asserts to fail on each platform.